pkg/operator/ceph/cluster/mon/spec.go

```diff
@@ -283,7 +304,6 @@ func (c *Cluster) makeMonDaemonContainer(monConfig *monConfig) v1.Container {
 			k8sutil.PodIPEnvVar(podIPEnvVar),
 		),
 		Resources: cephv1.GetMonResources(c.spec.Resources),
-		Lifecycle: opspec.PodLifeCycle(""),
```

Comment:
Do we need to consider this change for other daemons? For now, maybe we should leave this line (but commented out) and add a comment about why it's not needed. Otherwise, someone might think it is missing since all the other daemons have it and put it back.

Reply:
Why would we only be seeing this now for mons on PVs? That would affect the data dir, but remind me why is the log dir affected as well?

Reply:
There are two issues here. According to the docs, using a post start life cycle for anything that the entry point depends on is a race. Assuming that's a correct reading and the chown is in fact a dependency, that's the primary reason I left this patch in.

The second issue is the problem we saw with `proc/<pid>/ipc` file not being found when chown was being run in the life cycle hook. I don't have a root cause for that, but it doesn't seem to show up the chown is performed in an init container.

A similar error here https://bugzilla.redhat.com/show_bug.cgi?id=1701326#c6 and someone mentioning a race between CRI-O and kubelet here https://github.com/cri-o/cri-o/issues/1927#issuecomment-452631660

Presumably postStart hooks are intended to observe containers that are fully initialized, but if anything would exacerbate an existing startup race, I think PVC attachments qualify.

Reply:
added a note on the lifecycle hook linking back here for the discussion